drivers/timer/stm32_lptim_timer.c

```diff
@@ -166,6 +166,23 @@ int z_clock_driver_init(struct device *device)
 	return 0;
 }
 
+static inline uint32_t z_clock_lptim_getcounter(void)
+{
+	uint32_t lp_time;
+	uint32_t lp_time_prev_read;
+
+	/* It should be noted that to read reliably the content
+	 * of the LPTIM_CNT register, two successive read accesses
+	 * must be performed and compared
+	 */
+	lp_time = LL_LPTIM_GetCounter(LPTIM1);
+	do {
+		lp_time_prev_read = lp_time;
+		lp_time = LL_LPTIM_GetCounter(LPTIM1);
+	} while (lp_time != lp_time_prev_read);
```

Comment:
could it be like:
```
uint32_t lp_time = LL_LPTIM_GetCounter(LPTIM1);
while (lp_time != LL_LPTIM_GetCounter(LPTIM1)) {
	lp_time = LL_LPTIM_GetCounter(LPTIM1);
}

```

Reply:
In my opinion using a while loop in this way works fine in the case you get the same vale in two reads. If the second read is different, its value is lost, then a third read is done in the loop body and a fourth in the loop test again. So we'll have four reads, when three should suffice.
The do-while loop is more convenient when you have something that has to be done one or more times (two reads at least). With my proposal the fourth read is not done. It requires a further local variable, but the compiler should just use a register for it.
The optimization is rather small (I didn't check the generated code yet), but I believe that even the logic flow is more natural using a do-while loop.
I tested it in my prototype using STM32L462 using k_sleep() with 600 ms and 1 sec and measuring the actual delta time using k_cycle_get_32(). Actual time is also indirectly checked by continuous messages received on serial line from a GNSS device without losses and by Teraterm timestamps.

Reply:
I agree. 
Originally this sequence was recommended from the lp_ticker_read(void) function of the https://github.com/ARMmbed/mbed-os/blob/master/targets/TARGET_STM/lp_ticker.c
It can be optimized, in this way